### PR TITLE
Article featured image display

### PR DESCRIPTION
**Pull request by joshdarby**
## Changes

This pull request makes the following changes:

- Creates the featured image header display for single article pages.

With image (desktop):
screencapture-workdayminn-test-the-mac-commission-gets-heated-2019-08-02-14_40_11

With image (mobile):
screencapture-workdayminn-test-the-mac-commission-gets-heated-2019-08-02-14_40_59

Without image (desktop):
screencapture-workdayminn-test-new-test-post-2019-08-02-14_41_22

Without image (mobile):
screencapture-workdayminn-test-new-test-post-2019-08-02-14_41_32

## Why

For #8 

## Testing/Questions

Features that this PR affects:

- Article page headers

Questions that need to be answered before merging:

- [ ] Does it look ok on desktop?
- [ ] Does it look ok on mobile?
- [ ] Anything that could be improved?

Steps to test this PR:

1. View a post on mobile and desktop that has a featured image
2. View a post on mobile and desktop that does not have a featured image

### CODE REVIEW

File: wp-content/themes/workday/inc/photo-header-template.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ * The inline style="" contains the background-image information for the photo header.
+ * The closing </header> tag is assumed to be present elsewhere in the markup.
+ * @param WP_Post|int the attachment image's post or ID
+ * @return string HTML
+ */
+function workday_photo_header_tag( $post = null ) {
+	$element_id = 'photo-header-' . (string) $post;
+	$selector = '#' . $element_id;
+
+	if ( $post ) {
```
Comment (benlk) [inline wp-content/themes/workday/inc/photo-header-template.php]: Would it be easier to say

```php
if ( ! is_a( 'WP_Post', $post ) ) {
    $post = get_post( $post );
}
```

to ensure that $post is set, and then go on with the output here, rather than wrapping half the logic in a "if there is a post"?

File: wp-content/themes/workday/less/vars.less [hunk 2]
```diff
@@ -2,6 +2,7 @@
 // if updating these colors, please update inc/block-color-palette.php
 @green: #71E26C; // for use on dark backgrounds;
 @darkgreen: #16A327; // for use on light backgrounds;
+@mint-green: #7BE2B6;
```
Comment (benlk) [inline wp-content/themes/workday/less/vars.less]: Should we use plain `@green` here?

File: wp-content/themes/workday/partials/content-single.php [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            }
+
+            ?>
+            <?php largo_maybe_top_term(); ?>
+            <h1 class="entry-title" itemprop="headline"><?php the_title(); ?></h1>
+            <h5 class="byline"><?php largo_byline( true, true, get_the_ID() ); ?></h5>
+
+            <?php if ( ! of_get_option( 'single_social_icons' ) == false ) { ?>
+                <ul class="photo-header-social-links">
+                    <?php largo_social_links(); ?>
```
Comment (benlk) [inline wp-content/themes/workday/partials/content-single.php]: Instead, should we use the largo_post_social_links function to output these links with sharing intent? https://github.com/INN/largo/blob/v0.6.4/inc/post-social.php#L26

File: wp-content/themes/workday/inc/photo-header-template.php (hunk 1 above)
Comment (benlk) [reply wp-content/themes/workday/inc/photo-header-template.php]: Eh; not an important question since we're the ones calling the function and making sure that its call always gets passed an ID.

File: wp-content/themes/workday/partials/content-single.php (hunk 3 above)
Comment (joshdarby) [reply wp-content/themes/workday/partials/content-single.php]: Probably. I wasn't aware of that one. I'll swap them out.

File: wp-content/themes/workday/less/vars.less (hunk 2 above)
Comment (joshdarby) [reply wp-content/themes/workday/less/vars.less]: In the design on Xd it's `#7BE2B6` so I'm not sure. @MirandaEcho?

For reference: https://xd.adobe.com/spec/e12ba5e6-9fe1-448a-68e4-e701056351e0-971a/screen/b31b36e7-9b99-4a04-b37a-572cb5815bf0/Article-Revised

Comment (benlk) [reply wp-content/themes/workday/less/vars.less]: Those mockups have like eight different shades of green.

File: wp-content/themes/workday/partials/content-single.php (hunk 3 above)
Comment (joshdarby) [reply wp-content/themes/workday/partials/content-single.php]: @benlk There's no way to filter the output list of `largo_post_social_links` so we can redo the order and also display LinkedIn and Pinterest, is there? We'll probably need to rebuild that function to suit the icons that are requested here.

Comment (benlk) [reply wp-content/themes/workday/partials/content-single.php]: No, there isn't. (Unless you feel like parsing HTML.) It's touched upon in https://github.com/INN/largo/issues/1708, for an undetermined future Largo version.

I think in other themes we've straight-up redefined `largo_post_social_links()` (it is a pluggable function) in order to add new social buttons or reorder them.

File: wp-content/themes/workday/less/vars.less (hunk 2 above)
Comment (MirandaEcho) [reply wp-content/themes/workday/less/vars.less]: Whatever is closest/close-ish to the mockups is fine.

File: wp-content/themes/workday/partials/content-single.php (hunk 3 above)
Comment (joshdarby) [reply wp-content/themes/workday/partials/content-single.php]: @MirandaEcho Before I go too far down the rabbit hole, is it ok if I do this?

Basically, we have 2 functions, one that displays links to the clients social media (which is the function I was using) and one that displays links *to share* the post on social media.

If we want to use function #2 so users can share the post from the buttons we will need to create a custom function based off of it since it does not show the correct social media sites

Comment (kaylima) [reply wp-content/themes/workday/partials/content-single.php]: @joshdarby it's fine to rearrange the social icons. as far as I understand, that's all we would need to filter for with the function that displays links to share. is that true? or is there further filtering needed.

TL;DR:
We don't have to follow the mockup exactly for the social icon display, so let's figure out a quick/easy way to use the sharing function.

Comment (joshdarby) [reply wp-content/themes/workday/partials/content-single.php]: @kaylima We would also need to add in functionality to share to Pinterest and LinkedIn.

**Comment by joshdarby** [comment]
@benlk Can you give this one more quick glance over? I swapped out `largo_social_links()` instead of `largo_post_social_links()`

**Comment by benlk** [comment]
Buttons look good, click handlers function, and it resizes.
LGTM. :shipit:
